app/src/hooks/useLayer2Balance.ts

```diff
@@ -0,0 +1,43 @@
+import { useAccount, useBalance as useBalanceL2 } from 'wagmi';
+import { useL2GasEstimate } from './useL2GasEstimate';
+import { MINIMUM_SEND_AMOUNT } from '../lib/constants';
+import { formatEther } from 'viem';
+
+const GENERIC_IOTA_ADDRESS = '0x1111111111111111111111111111111111111111111111111111111111111111';
+
+export function useLayer2Balance(): {
+    availableBalance: bigint;
+    isLoading: boolean;
+    formattedAvailableBalance: string;
+} {
+    const layer2Account = useAccount();
+
+    // Fetch Layer 2 balance
+    const { data: layer2BalanceData, isLoading: isLoadingL2 } = useBalanceL2({
+        address: layer2Account?.address as `0x${string}`,
+        query: {
+            refetchInterval: 2000, // Refetch Layer 2 balance every 2 seconds
+        },
+    });
+
+    const layer2TotalBalance = layer2BalanceData?.value || 0n;
+
+    const { data: gasEstimationData, isPending: isGasEstimationLoading } = useL2GasEstimate({
+        address: GENERIC_IOTA_ADDRESS,
+        amount: MINIMUM_SEND_AMOUNT.toString(),
```

Comment:
Are we sure doing the estimate with always the same hardcoded `0.1` amount is precise enough ?
As far as i see changing this amount changes the estimate we get. I suppose optimally we would use the amount that the user wants to send, but i understand that we possibly don't have it at this point.